packages/tools/devtools/devtools-core/src/data-visualization/DataEditing.ts

```diff
@@ -0,0 +1,43 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISharedObject } from "@fluidframework/shared-object-base";
+import { Serializable } from "@fluidframework/datastore-definitions";
+import { EditType } from "../CommonInterfaces";
+
+// Ideas:
+// - Hold onto previous summary and only transmit diff?
+
+// TODOs:
+// - Dependency tracking
+//   - When a particular DDS is no longer reachable via the input data, we need to remove it from the map and stop
+//     emitting updates.
+
+/**
+ * The type of a shared object.
+ *
+ * @remarks
+ *
+ * This can be acquired via {@link @fluidframework/datastore-definitions#IChannelFactory.Type} field of
+ * your shared object's factory class.
+ */
+export type SharedObjectType = string;
+
+/**
+ * Generates a visual description of the provided {@link @fluidframework/shared-object-base#ISharedObject}'s
+ * current state.
+ *
+ * @param sharedObject - The object whose data will be rendered.
+ * @param visualizeChildData - Callback to render child content of the shared object.
+ *
+ * @returns A visual tree representation of the provided `sharedObject`.
+ *
+ * @public
+ */
+export type EditSharedObject = (
+	sharedObject: ISharedObject,
+	data: Serializable,
```

Comment:
What do you think about replacing "data" and "type" with a parameter called "edit" (or something) that is an object with the "data" and "type" properties? In terms of backwards and forwards compatibility, it is much easier to change the properties of a type/interface than it is to change the parameters of a function signature.

Reply:
Should those properties be optional, in theory they are necessary but they might allow a bit more forward/backward compat?

Reply:
Made new changes, let me know if they are inline with what you think it should be like. (And what do you think about the comment above?)